src/rust/engine/fs/store/src/lib.rs

```diff
@@ -773,21 +773,29 @@ impl Store {
   ) -> BoxFuture<(), String> {
     let store = self.clone();
     async move {
-      if let RootOrParentMetadataBuilder::Root(..) = root_or_parent_metadata {
-        let destination = destination.clone();
-        store
-          .local
-          .executor()
-          .spawn_blocking(move || fs::safe_create_dir_all(&destination))
-          .await?;
-      } else {
-        let destination = destination.clone();
-        store
-          .local
-          .executor()
-          .spawn_blocking(move || fs::safe_create_dir(&destination))
-          .await?;
-      }
+      let directory_creation =
+        if let RootOrParentMetadataBuilder::Root(..) = root_or_parent_metadata {
+          let destination = destination.clone();
+          store
+            .local
+            .executor()
+            .spawn_blocking(move || fs::safe_create_dir_all(&destination))
+            .await
+        } else {
+          let destination = destination.clone();
+          store
+            .local
+            .executor()
+            .spawn_blocking(move || fs::safe_create_dir(&destination))
+            .await
+        };
+      directory_creation.map_err(|e| {
```

Comment:
FTR: This had nothing to do with the fix: was just useful for tracking down the issue.